kratos/python/add_geometries_to_python.cpp

```diff
@@ -153,22 +115,50 @@ void  AddGeometriesToPython(pybind11::module& m)
     .def("Dimension", &GeometryType::Dimension)
     .def("DomainSize",&GeometryType::DomainSize)
     .def("PointsNumber",&GeometryType::PointsNumber)
+    // Integration
+    .def("IntegrationPointsNumber", [](GeometryType& self)
+        { return(self.IntegrationPointsNumber()); })
     // Quadrature points
-    .def("CreateQuadraturePointGeometries", CreateQuadraturePointGeometries1)
-    .def("Normal",GetNormal)
-    .def("Normal",FastGetNormal)
-    .def("UnitNormal",GetUnitNormal)
-    .def("UnitNormal",FastGetUnitNormal)
+    .def("CreateQuadraturePointGeometries", [](GeometryType& self,
+        GeometriesArrayType& rResultGeometries, IndexType NumberOfShapeFunctionDerivatives)
+        { return(self.CreateQuadraturePointGeometries(rResultGeometries, NumberOfShapeFunctionDerivatives)); })
+    // Normal
+    .def("Normal", [](GeometryType& self)
+        { const auto& r_type = self.GetGeometryType(); KRATOS_WARNING_IF("Geometry", !(r_type == GeometryData::KratosGeometryType::Kratos_Line2D2 || r_type == GeometryData::KratosGeometryType::Kratos_Triangle3D3)) << "WARNING:: Your geometry is not linear, please provide local coordinates or a GP index. Normal will be computed in zero local coordinates" << std::endl; CoordinatesArrayType LocalCoords; LocalCoords.clear(); return(self.Normal(LocalCoords)); })
+    .def("Normal", [](GeometryType& self, CoordinatesArrayType& LocalCoords)
+        { return(self.Normal(LocalCoords)); })
+    .def("Normal", [](GeometryType& self, IndexType IntegrationPointIndex)
+        { return(self.Normal(IntegrationPointIndex)); })
+    .def("UnitNormal", [](GeometryType& self)
+        { const auto& r_type = self.GetGeometryType(); KRATOS_WARNING_IF("Geometry", !(r_type == GeometryData::KratosGeometryType::Kratos_Line2D2 || r_type == GeometryData::KratosGeometryType::Kratos_Triangle3D3)) << "WARNING:: Your geometry is not linear, please provide local coordinates or a GP index. Normal will be computed in zero local coordinates" << std::endl; CoordinatesArrayType LocalCoords; LocalCoords.clear(); return(self.UnitNormal(LocalCoords)); })
```

Comment:
this is where lambdas are not really that useful any more, one can barely read what is happening ;)

```suggestion
        { const auto& r_type = self.GetGeometryType(); KRATOS_WARNING_IF("Geometry", !(r_type == GeometryData::KratosGeometryType::Kratos_Line2D2 || r_type == GeometryData::KratosGeometryType::Kratos_Triangle3D3)) << "Your geometry is not linear, please provide local coordinates or a GP index. Normal will be computed in zero local coordinates" << std::endl; CoordinatesArrayType LocalCoords; LocalCoords.clear(); return(self.UnitNormal(LocalCoords)); })
```